lib/chat_mailer.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class DiscourseChat::ChatMailer
+  def send_unread_mentions_summary
+    return unless SiteSetting.chat_enabled
+
+    users.find_each do |user|
+      Jobs.enqueue(:user_email,
+        type: "chat_summary",
+        user_id: user.id,
+        force_respect_seen_recently: true
+      )
+
+      user.user_option.update(last_emailed_for_chat: Time.now)
+    end
+  end
+
+  def users
+    User
+      .distinct
+      .joins(:user_option)
+      .where(user_options: { chat_enabled: true, chat_email_frequency: UserOption.chat_email_frequencies[:when_away] })
+      .where('user_options.last_emailed_for_chat IS NULL OR user_options.last_emailed_for_chat < ?', 5.minutes.ago)
```

Comment:
Why 5 minutes ago?

Reply:
We should rely on `user#last_seen_at` here. No need to keep this field.